tests/test_apel_plugin.py

```diff
@@ -21,3 +25,90 @@ async def test_regex_dict_lookup(self):
             await functions.regex_dict_lookup(term_c, dict)
         assert pytest_error.type == SystemExit
         assert pytest_error.value.code == 1
+
+    async def test_get_begin_previous_month(self):
+        time_a = datetime(2022, 10, 23, 12, 23, 55)
+        time_b = datetime(1970, 1, 1, 00, 00, 00)
+
+        result = await functions.get_begin_previous_month(time_a)
+        assert result == datetime(2022, 9, 1, 00, 00, 00, tzinfo=pytz.utc)
+
+        result = await functions.get_begin_previous_month(time_b)
+        assert result == datetime(1969, 12, 1, 00, 00, 00, tzinfo=pytz.utc)
+
+    async def test_sql_filter(self):
+        create_table_sql = """
+                           CREATE TABLE IF NOT EXISTS records(
+                           site TEXT NOT NULL,
+                           year INTEGER NOT NULL,
+                           month INTEGER NOT NULL
+                           )
+                           """
+
+        insert_record_sql = """
+                            INSERT INTO records(
+                               site,
+                               year,
+                               month
+                           )
+                           VALUES(
+                               ?, ?, ?
+                           )
+                           """
+
+        conn = await aiosqlite.connect(":memory:")
```

Comment:
- random order: shouldn't matter, since I open and close the connection to the database inside of each test, the latter causes the deletion of the in-memory database
- concurrently: I don't think that happens, since people are asking how to do this and the answer is to use a plugin

Reply:
concurrently is not the same as in parallel. Since those tests are async, they run in an async scheduler which gets control of a thread back at every await point. The scheduler could decide at an await point in test A to progress test B. Assuming all threads run in the same runtime, it could happen that a database connection is held across an await point, potentially having two tests writing to the same in-memory database. Obviously I'm not an expert on how this works exactly, but if you do see spuriously failing tests, this may be something to consider.